Add `cmd-delete-word-back` command

janpeterd opened the pull request:
Related issues:
-  #281 
- #1406

A new command `cmd-delete-word-back` is added. 
~~A new configuration option `unixwords` is added.~~
Docs about this command added.

This is my first pull request and I am inexperienced in `go`. So any feedback or suggestions are welcome.

joelim-work commented:
@janpeterd Thanks for your contribution. I am not sure if you have seen this yet, but there is a set of [guidelines](https://github.com/gokcehan/lf/wiki/Contributing) for first-time contributors.

Regarding your changes, I think it is better to not add the `unixwords` option at all, and just providing a separate `cmd-delete-word-back` command should suffice. The relationship between `cmd-delete-word-back` and and `cmd-delete-word` should be similar to the relationship between `cmd-word-back` and `cmd-word`.

BTW I think the line editing in `lf` is based on `bash`, and what you are really trying to do is to implement `backward-kill-word` (press `alt+backspace`) in https://www.gnu.org/software/bash/manual/html_node/Commands-For-Killing.html

One other thing I noticed is that you have saved the deleted text in the yank buffer for `cmd-delete-word-back`, but interestingly `cmd-delete-word` doesn't do this. I'm not sure which one is correct but I think the behavior should be consistent between these two commands.

gokcehan commented:
@janpeterd Thank you very much for working on this.

@joelim-work I don't remember anything about the missing yank for `cmd-delete-word`. It is probably a mistake so we should probably fix it. I will try to go over all command-line commands to make sure they have compatible behavior with readline/bash defaults.

@janpeterd I agree with @joelim-work in that `unixwords` option is not necessary. Support for `alt+backspace` in `lf` was added recently. So maybe we can use it for this new `cmd-delete-word-back` command as this shortcut is mentioned in both bash and readline documentations.

Not quite related to this discussion, but I'm thinking of moving the Contributions wiki page to the repository. I think it is more common to find it in the repository. It is also mostly intended for developers so it should be separate from the content intended for users (i.e. wiki).

janpeterd commented:
I removed the configuration option and mapped the command by default to `<a-backspace>` and `<a-backspace2>`. Also `bash` doesn't save the removed text to the yank-buffer for this command, so I have removed that code. Thanks for bringing that up @joelim-work.

joelim-work commented on doc.go:
```diff
@@ -111,6 +111,7 @@ The following command line commands are provided by lf:
 	cmd-word                 (default '<a-f>')
 	cmd-word-back            (default '<a-b>')
 	cmd-delete-word          (default '<a-d>')
+	cmd-delete-word-back     (default '<a-backspace> and <a-backspace2>')
```
I think you made a small typo:

```suggestion
	cmd-delete-word-back     (default '<a-backspace>' and '<a-backspace2>')
```

joelim-work commented on opts.go:
```diff
@@ -354,6 +354,8 @@ func init() {
 	gOpts.cmdkeys["<a-u>"] = &callExpr{"cmd-uppercase-word", nil, 1}
 	gOpts.cmdkeys["<a-l>"] = &callExpr{"cmd-lowercase-word", nil, 1}
 	gOpts.cmdkeys["<a-t>"] = &callExpr{"cmd-transpose-word", nil, 1}
+	gOpts.cmdkeys["<a-backspace>"] = &callExpr{"cmd-delete-word-back", nil, 1}
+	gOpts.cmdkeys["<a-backspace2>"] = &callExpr{"cmd-delete-word-back", nil, 1}
```
Very minor, but I would suggest to move this to just under `cmd-delete-word` to group them together.

joelim-work reviewed (commented):
Looks mostly good, just a couple of very minor comments.

joelim-work commented:
@gokcehan I don't have any issues with moving the contribution guidelines to `CONTRIBUTING.md`, it sounds good to me.

janpeterd commented:
@joelim-work typo corrected and block of code moved.

gokcehan commented:
Looks good to me as well. Thank you @janpeterd for the patch and @joelim-work for the review.

By the way, both <kbd>alt+d</kbd> and <kbd>alt+backspace</kbd> save the deleted text in the yank buffer on my bash. I can see these commands are listed under "Killing and Yanking" in `man readline`,  so I think this is the expected behavior. Therefore, I'm thinking of adding yanking to both of these commands in `lf` as well. Let me know if there is an issue with this.
